src/Framework/Telemetry/BuildTelemetry.cs

```diff
@@ -167,5 +167,56 @@ public override IDictionary<string, string> GetProperties()
 
             return properties;
         }
+
+        /// <summary>
+        /// Create a list of properties sent to VS telemetry with the information whether they should be hashed.
+        /// </summary>
+        /// <returns></returns>
+        public IList<TelemetryItem> GetActivityProperties()
```

Comment:
A suggestion, feel free to ignore:
can we logically split the types of collected data? e.g. `AddFeatureFlags`, `AddBuildProperties`, `AddTimeMetrics`, to improve readability and simplify adding new lines .

It looks like this method has a huge potential to grow :)

Reply:
My idea is that this class is holding basic datapoints that are shared between sdk and VS telemetry which won't change soon, so likely YAGNI.
For more telemetry events with multiple data points there will be different classes holding that data implementing IActivityTelemetryDataHolder. I am open to the opinions that this scheme is wrong in it's entirety, but if not I think refactoring this specific class is overengineering.